### PR TITLE
castget: init at 1_2_4

### PR DESCRIPTION
**Pull request by doronbehar**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by doronbehar** [comment]
Bump.

**Comment by doronbehar** [comment]
Good bot.

File: pkgs/applications/networking/feedreaders/castget/default.nix
Comment (timokau) [inline pkgs/applications/networking/feedreaders/castget/default.nix]: I'd prefer having the version in the conventional point-separated style. For the `rev` you can then use `lib.replaceStrings ["."] ["-"] version`.

File: pkgs/applications/networking/feedreaders/castget/default.nix
Comment (timokau) [inline pkgs/applications/networking/feedreaders/castget/default.nix]: Why is this necessary?

File: pkgs/applications/networking/feedreaders/castget/default.nix
Comment (timokau) [inline pkgs/applications/networking/feedreaders/castget/default.nix]: Are you prepared to maintain this? If so, please add yourself as a maintainer.

File: pkgs/applications/networking/feedreaders/castget/default.nix
Comment (timokau) [inline pkgs/applications/networking/feedreaders/castget/default.nix]: This style is pretty unconventional. If you have a multi-line arglist, you usually put the final `}:` on a separate line. Personally I prefer to just make one line per dependency, but there is no clear consensus on that.

**Review by timokau — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/feedreaders/castget/default.nix
Comment (doronbehar) [reply pkgs/applications/networking/feedreaders/castget/default.nix]: Otherwise, `autoreconf` fails with:

```
Makefile.am: error: required file './NEWS' not found
Makefile.am: error: required file './README' not found
Makefile.am: error: required file './ChangeLog' not found
```

File: pkgs/applications/networking/feedreaders/castget/default.nix
Comment (doronbehar) [reply pkgs/applications/networking/feedreaders/castget/default.nix]: Fixed.

File: pkgs/applications/networking/feedreaders/castget/default.nix
Comment (doronbehar) [reply pkgs/applications/networking/feedreaders/castget/default.nix]: Of course :) Added.

File: pkgs/applications/networking/feedreaders/castget/default.nix
Comment (doronbehar) [reply pkgs/applications/networking/feedreaders/castget/default.nix]: Fixed.

File: pkgs/applications/networking/feedreaders/castget/default.nix
Comment (timokau) [reply pkgs/applications/networking/feedreaders/castget/default.nix]: Weird that they don't mention that in the README. Can you replace the `touch` by `automake --add-missing` to be more robust and better match the intent? Also, this should be done in `preAutoreconf` instead of `prePatch` since its an extension to the autoreconf phase.

Comment (doronbehar) [reply pkgs/applications/networking/feedreaders/castget/default.nix]: Did you mean to run `automake --add-missing` in the `preAutoreconf` hook? That fails with the same errors. It seems from the log that this command is run with these flags by the autoreconf hook itself:

```
autoreconf: running: /nix/store/0yl3mx4f8f1bnmyfzxv6a3bik7c2ni87-autoconf-2.69/bin/autoconf --force
autoreconf: running: /nix/store/0yl3mx4f8f1bnmyfzxv6a3bik7c2ni87-autoconf-2.69/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
```

Using the same `touch` commands but only using `preAutoreconf` instead of `prePatch` works as well.

From AFAIK, the requirement for these files is a GNU strictness that probably stayed there from the old days this software was distributed by them. The release is from 2017 so I'm not surprised it's not documented.

See this: https://www.gnu.org/software/automake/manual/automake.html#index-gnu-strictness

Comment (timokau) [reply pkgs/applications/networking/feedreaders/castget/default.nix]: That's weird, especially since castget is marked as `foreign`. Anyways, its not that important.

**Review by timokau — APPROVED** [review]
LGTM, will merge when ofBorg approves. Thank you!

**Comment by doronbehar** [comment]
Thank you! As a general question, I was wondering, is it usually desired by maintainers that packages related PRs will consist of a single commit? `@worldofpeace` has asked me in the past to squash changes and since then I've always made changes with a `git commit --amend --no-edit` and pushed with `--force`.

**Comment by timokau** [comment]
It depends on context (but also a lot on the reviewer's preferences). My opinion is that every commit should do one thing, but there is no reason to maintain all histories within a PR. So in this case, the "one thing" is initializing a package. Nobody gains any information from the additional commits incorporating my nitpicks (and they don't have the appropriate naming scheme anyway), so I've used github's squash-and-merge feature.

But if, for example, this was the first package you maintain you should have one commit adding yourself to the maintainers list and then a separate commit init'ing the package (using the maintainers entry). Or if you're fixing a broken package and while you're at it also updating it to the latest version, you could have two commits: `package: fix build` and `package: old_version -> new_version`.

Basically two rules of thumb:
- you shouldn't need to use the word "and" in the commit message
- ask yourself if it could ever be useful to have these as separate commits (for example I might want to revert the init of a package, but that shouldn't revert the addition to the maintainers list as well)

And remember that these are rules of thumb and if its easier to just do everything in one commit, sometimes its just not worth to spend the time and dis-entangle everything ;)

**Pull request merged**
